process/rag_data_test.py

```diff
@@ -0,0 +1,59 @@
+from datasets import load_dataset
+from langchain_core.prompts import ChatPromptTemplate
+from langchain_community.vectorstores import Chroma
+from langchain_community.llms import Ollama
+from langchain.chains import create_retrieval_chain
+from langchain_community.embeddings.ollama import OllamaEmbeddings
+import pandas as pd
+from langchain.chains.combine_documents import create_stuff_documents_chain
+import chromadb
+
+dataset = load_dataset("neural-bridge/rag-dataset-12000", split="train")
+dataset = pd.DataFrame(dataset)[:50]
+
+class DefChromaEF(OllamaEmbeddings):
+  def __init__(self,ef):
+    self.ef = ef
+    self.model = "phi3"
+
+  def embed_documents(self,texts):
+    return self.ef(texts)
+
+  def embed_query(self, query):
+    return self.ef([query])[0]
+
+chroma_client = chromadb.Client()
+collection = chroma_client.create_collection(name='rag_data', embedding_function=DefChromaEF.embed_documents)
```

Comment:
- Please check out this [link](https://stackoverflow.com/questions/76650513/dynamically-add-more-embedding-of-new-document-in-chroma-db-langchain) to learn more about dynamically adding more embeddings of new documents in Chroma DB.

- We might need to move the embedding code block to a separate module instead of keeping everything in one place. This way, we can avoid re-embedding and storing the data in Chroma DB repeatedly.